app/src/main/java/org/oppia/android/app/profile/ResetPinDialogFragmentPresenter.kt

```diff
@@ -45,20 +46,23 @@ class ResetPinDialogFragmentPresenter @Inject constructor(
       viewModel = resetViewModel
     }
 
-    binding.inputPin.setInput(resetViewModel.inputPin.get().toString())
-    binding.inputPin.addTextChangedListener(object : TextWatcher {
-      override fun onTextChanged(confirmPin: CharSequence?, start: Int, before: Int, count: Int) {
-        confirmPin?.let {
-          resetViewModel.inputPin.set(confirmPin.toString())
+    // [onTextChanged] is a extension function defined at [TextInputEditTextHelper]
+    binding.resetPinInputPinEditText.onTextChanged { confirmPin ->
+      confirmPin?.let {
+        if (
+          resetViewModel.errorMessage.get()?.isNotEmpty()!! && resetViewModel.inputPin.get() == it
+        ) {
+          resetViewModel.inputPin.set(it)
+        } else {
+          resetViewModel.inputPin.set(it)
           resetViewModel.errorMessage.set("")
         }
       }
-
-      override fun afterTextChanged(confirmPin: Editable?) {}
-      override fun beforeTextChanged(p0: CharSequence?, start: Int, count: Int, after: Int) {}
-    })
-
-    binding.inputPin.setLabel("$name's New PIN")
+    }
+    // This is set dynamically as the name is not passed to [ResetPinViewModel]
```

Comment:
Need more understandable comment that why we are setting it. What will crash if we do through XML.

Reply:
@anandwana001  does this look good
```
// Hint is set dynamically cause it is not possbile to use string formatting using xml as the name is not available in the viewmodel
```

Reply:
@anandwana001 Or can we try to set the hint in the xml by passing the name in the `ResetPinViewModel`

Reply:
we can set in XML directly, but as you had mentioned earlier it will cause some issue with configuration change?
if it was resolved then let's set in XML only.

Reply:
Actually, that was for error message I've updated it